lib/FNAMF/main.c

```diff
@@ -356,19 +355,20 @@ INT CDECL tf_readaudio(struct context *ctx, float *dst, int dst_count)
     }
 
     if (FAILED(hr))
-        dst_count = -1;
+        ret_count = -1;
     else if (!ctx->audio_buffer || ctx->eos)
-        dst_count = 0;
+        ret_count = 0;
     else
     {
+        int rem_count;
         hr = IMFMediaBuffer_Lock(ctx->audio_buffer, (BYTE **)&src, NULL, &len);
         if (SUCCEEDED(hr))
         {
-            dst_count = min(len / sizeof(float), dst_count);
+            ret_count = min(len / sizeof(float), dst_count);
             memcpy(dst, src, dst_count * sizeof(float));
-            len -= dst_count * sizeof(float);
+            len -= ret_count * sizeof(float);
 
-            memcpy(src, src + dst_count, len);
+            memcpy(src, src + ret_count, len);
```

Comment:
I think the memory regions can overlap here so it should be `memmove`.

Reply:
Good catch. I've changed to `memmove`.